lib/rules/unicode-bom/index.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+const report = require("../../utils/report");
+const ruleMessages = require("../../utils/ruleMessages");
+const validateOptions = require("../../utils/validateOptions");
+
+const ruleName = "unicode-bom";
+
+const messages = ruleMessages(ruleName, {
+  expected: "Expected Unicode BOM",
+  rejected: "Unexpected Unicode BOM"
+});
+
+const rule = function(expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: ["always", "never"]
+    });
+
+    if (
+      !validOptions ||
+      root.source.inline ||
+      root.source.lang === "object-literal" ||
+      root.document !== undefined
```

Comment:
I have some doubts about this check, but I have not found another way to verify that it is not a HTML document.

Reply:
Could you add a comment into the code explaining what this condition checks, please.